recipe/magento2.php

```diff
@@ -97,17 +122,71 @@
 
 desc('Deploys assets');
 task('magento:deploy:assets', function () {
-
     $themesToCompile = '';
-    if (count(get('magento_themes')) > 0) {
+    if (get('split_static_deployment')) {
+        invoke('magento:deploy:assets:adminhtml');
+        invoke('magento:deploy:assets:frontend');
+    } elseif (count(get('magento_themes')) > 0 ) {
         foreach (get('magento_themes') as $theme) {
             $themesToCompile .= ' -t ' . $theme;
         }
+        run("{{bin/php}} {{release_or_current_path}}/bin/magento setup:static-content:deploy --content-version={{content_version}} {{static_deploy_options}} {{static_content_locales}} $themesToCompile -j {{static_content_jobs}}");
     }
+});
+
+desc('Deploys assets for backend only');
+task('magento:deploy:assets:adminhtml', function () {
+    magentoDeployAssetsSplit('backend');
+});
 
-    run("{{bin/php}} {{release_or_current_path}}/bin/magento setup:static-content:deploy --content-version={{content_version}} {{static_content_locales}} $themesToCompile -j {{static_content_jobs}}");
+desc('Deploys assets for frontend only');
+task('magento:deploy:assets:frontend', function () {
+    magentoDeployAssetsSplit('frontend');
 });
 
+/**
+ * @phpstan-param 'frontend'|'backend' $area
+ *
+ * @throws ConfigurationException
+ */
+function magentoDeployAssetsSplit(string $area)
+{
+    if (!in_array($area, ['frontend', 'backend'], true)) {
+        throw new ConfigurationException("\$area must be either 'frontend' or 'backend', '$area' given");
+    }
+
+    $isFrontend = $area === 'frontend';
+    $suffix = $isFrontend
+        ? ''
+        : '_backend';
+
+    $themesConfig = get("magento_themes$suffix");
+    $defaultLanguages = get("static_content_locales$suffix");
+    $useDefaultLanguages = array_is_list($themesConfig);
+
+    /** @var list<string> $themes */
+    $themes = $useDefaultLanguages
+        ? array_values($themesConfig)
+        : array_keys($themesConfig);
+
+    $staticContentArea = $isFrontend
+        ? 'frontend'
+        : 'adminhtml';
+
+    if ($useDefaultLanguages) {
+        $themes = implode('-t ', $themes);
+
+        run("{{bin/php}} {{release_or_current_path}}/bin/magento setup:static-content:deploy --area=$staticContentArea --content-version={{content_version}} {{static_deploy_options}} $defaultLanguages $themes -j {{static_content_jobs}}");
```

Comment:
Should we add `--force` here by default? Otherwise deploy mode has to be "production" and on a build server you usually don't have that configured

Reply:
I think this makes sense

Reply:
The implode should be also reviewed.
If I have this:

```
array (
  0 => 'Hyva/default',
  1 => 'Magento/luma',
)
```

The implode makes this:

`Hyva/default-t Magento/luma`


However it should be this:

`-t Hyva/default -t Magento/luma`

Reply:
the implode is updated and the force option is added as well.

Reply:
@akosglue great, could you also fix the merge conflicts?

Reply:
seems to be resolved now

Reply:
@peterjaap any chance to merge this?